src/graph/executor/query/JoinExecutor.h

```diff
@@ -33,7 +33,13 @@ class JoinExecutor : public Executor {
 
   std::unique_ptr<Iterator> lhsIter_;
   std::unique_ptr<Iterator> rhsIter_;
+  // colSize_ is the size of the output columns of join executor.
+  // If the join is natural join, which means the left and right columns have duplicated names,
+  // the output columns will be a intersection of the left and right columns.
   size_t colSize_{0};
+  // If the join is natural join, rhsOutputColIdxs_ is used to record the output column index of the
+  // right
+  std::unique_ptr<std::vector<size_t>> rhsOutputColIdxs_;
```

Comment:
`std::vector<size_t>>` directly.

Reply:
If it is null, we should directly output the entire row of the right-side input. Otherwise, we should only output the non-duplicated columns of right-side input. And the non-duplicatd column index are recorded in the rhsOutputColIdxs_.

Reply:
it seems you could do this by checking whether the vector object is empty

Reply:
ok